### PR TITLE
Restructure the code to be extensible to different query modes, table types, file formats, etc.

### PR DESCRIPTION
**Pull request by yihua**

### CODE REVIEW

File: plugin/trino-hudi/src/main/java/io/trino/plugin/hudi/query/HudiFileListing.java [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+    public abstract List<FileStatus> listStatus(HudiPartitionInfo partitionInfo);
+
+    public void close()
+    {
+        if (!fileSystemView.isClosed()) {
+            fileSystemView.close();
+        }
+    }
+
+    protected void maybeInitFileSystemViewAndPredicates()
```
Comment (codope) [inline plugin/trino-hudi/src/main/java/io/trino/plugin/hudi/query/HudiFileListing.java]: Simply name `initFileSystemViewAndPredicates`?

File: plugin/trino-hudi/src/main/java/io/trino/plugin/hudi/HudiSplitManager.java [hunk 2]
```diff
@@ -76,26 +67,17 @@ public ConnectorSplitSource getSplits(
         SchemaTableName tableName = hudiTable.getSchemaTableName();
         HudiMetadata hudiMetadata = transactionManager.get(transaction);
         HiveMetastore metastore = hudiMetadata.getMetastore();
-        List<HiveColumnHandle> partitionColumnHandles = hudiMetadata.getColumnHandles(session, tableHandle)
+        Map<String, HiveColumnHandle> partitionColumnHandles = hudiMetadata.getColumnHandles(session, tableHandle)
```
Comment (codope) [inline plugin/trino-hudi/src/main/java/io/trino/plugin/hudi/HudiSplitManager.java:70]: Why do we need a map here? Won't the HiveColumnHandle contain the name?

**Review by codope — COMMENTED** [review]
@yihua Took one pass. Looks good to me. Couple of nits and if we can change the log statements to debug mode then we're good.

File: plugin/trino-hudi/src/main/java/io/trino/plugin/hudi/query/HudiFileListing.java (hunk 1 above)
Comment (yihua) [reply plugin/trino-hudi/src/main/java/io/trino/plugin/hudi/query/HudiFileListing.java]: Fixed.

File: plugin/trino-hudi/src/main/java/io/trino/plugin/hudi/HudiSplitManager.java (hunk 2 above)
Comment (yihua) [reply plugin/trino-hudi/src/main/java/io/trino/plugin/hudi/HudiSplitManager.java:70]: If there are more than one partition column, the order of column handles in the list matters.  `hudiMetadata.getColumnHandles()` may not have the same order for the partition columns as `table.getPartitionColumns()` so I put a mapping here for easier enforcement of the order in `HudiSplitSource`.

**Review by codope — APPROVED** [review]

**Pull request merged**